pallets/identity/src/claims.rs

```diff
@@ -481,6 +481,16 @@ impl<T: Config> Module<T> {
         // Sender has to be part of CDDProviders
         Self::ensure_authorized_cdd_provider(cdd_did)?;
 
+        let record = <DidRecords<T>>::get(cdd_did);
+        let cost = secondary_keys.iter().fold(0usize, |cost, auth| {
+            //Check permissions limits
+            Self::ensure_perms_length_limited(&auth.permissions).unwrap();
+            cost.saturating_add(auth.permissions.complexity())
+        });
```

Comment:
This is how to use `try_fold` to correctly handle the result from `ensure_perms_length_limited`:
```suggestion
        let cost = secondary_keys.iter().try_fold(0usize, |cost, auth| -> Result<usize, DispatchError> {
            //Check permissions limits
            Self::ensure_perms_length_limited(&auth.permissions)?;
            Ok(cost.saturating_add(auth.permissions.complexity()))
        })?;
```

Using `.unwrap()` there is completely wrong, since it turns the error into a panic.

Reply:
I understand, I made the recommended change thanks for your insight.